woocommerce/woocommerce-android
# Analytics outdated configuration

Closes: #11189 

### Why
The M2 configuration contains fewer cards than the M3. Merchants can have an outdated configuration from M2 if they have already configured the analytic cards they want in the Analytic Hub screen, so when releasing M3, we need to take this into account and create a new configuration based on the outdated M2 configuration and complete the new card's configuration with the default values.

### Description
This PR handles the edge-case scenario of a merchant with an outdated configuration opening the M3 version. For such cases, we will merge the outdated configuration with the default values of the missing cards and save this new configuration locally. 

### Testing instructions
1. Open the version from `trunk`
2. Tap on See all store analytics
3. Tap on the pencil at the top of the screen to enter the Customize Analytics screen
4. Update the card's configuration (visible cards/ order) and save the changes
5. Open the version from this branch
6. Tap on See all store analytics
7. Check that cards from M3 are displayed
8. Check that the configuration maintains all the settings from the previous configuration (visible cards/ order)
